boulder/src/draft/licenses.rs

```
@@ -0,0 +1,168 @@
+// SPDX-FileCopyrightText: Copyright © 2020-2025 Serpent OS Developers
+//
+// SPDX-License-Identifier: MPL-2.0
+
+use jwalk::WalkDir;
+use rapidfuzz::distance::levenshtein;
+use rayon::prelude::*;
+use std::collections::HashSet;
+use std::path::Path;
+use std::path::PathBuf;
+use tui::Styled;
+
+pub type Error = Box<dyn std::error::Error>;
+
+fn collect_spdx_licenses(dir: &Path) -> Result<(HashSet<PathBuf>, HashSet<PathBuf>), Error> {
+    // Collect our spdx licenses to compare against ensuring we don't match against deprecated licenses.
+    let mut purified_spdx_licenses = HashSet::new();
+    let spdx_license_paths: HashSet<_> = std::fs::read_dir(dir)?
+        .filter_map(|entry| {
+            entry.ok().and_then(|e| {
```

I don't think io error should be swallowed here (via `.ok()` + `filter_map`). Declaring `spdx_license_paths` as mutable beforehand and inserting into it in a `for` loop (just like `purified_spdx_licenses`) is probably the easiest / most readable way of allowing short-circuiting while iterating. (via `let entry = entry?;` as the first statement in the loop)